src/ssh/azext_ssh/resource_graph_utils.py

```diff
@@ -0,0 +1,114 @@
+
+
+from azure.cli.core._profile import Profile
+from azure.core.exceptions import HttpResponseError
+from azure.cli.core.azclierror import BadRequestError, AzureInternalError
+from knack.log import get_logger
+from knack.util import todict
+
+from .vendored_sdks.resourcegraph.models import ResultTruncated
+from .vendored_sdks.resourcegraph import ResourceGraphClient
+from .vendored_sdks.resourcegraph.models import \
+    QueryRequest, QueryRequestOptions, QueryResponse, ResultFormat, Error
+
+__SUBSCRIPTION_LIMIT = 1000
+__MANAGEMENT_GROUP_LIMIT = 10
+__logger = get_logger(__name__)
+
+
+def execute_query(client, graph_query, first, skip, subscriptions, management_groups, allow_partial_scopes, skip_token):
+    # type: (ResourceGraphClient, str, int, int, list[str], list[str], bool, str) -> object
+    mgs_list = management_groups
+    if mgs_list is not None and len(mgs_list) > __MANAGEMENT_GROUP_LIMIT:
+        mgs_list = mgs_list[:__MANAGEMENT_GROUP_LIMIT]
+        warning_message = "The query included more management groups than allowed. "\
+                          "Only the first {0} management groups were included for the results. "\
+                          "To use more than {0} management groups, "\
+                          "see the docs for examples: "\
+                          "https://aka.ms/arg-error-toomanysubs".format(__MANAGEMENT_GROUP_LIMIT)
+        __logger.warning(warning_message)
+
+    subs_list = None
+    if mgs_list is None:
+        subs_list = subscriptions or _get_cached_subscriptions()
+        if subs_list is not None and len(subs_list) > __SUBSCRIPTION_LIMIT:
+            subs_list = subs_list[:__SUBSCRIPTION_LIMIT]
+            warning_message = "The query included more subscriptions than allowed. "\
+                              "Only the first {0} subscriptions were included for the results. "\
+                              "To use more than {0} subscriptions, "\
+                              "see the docs for examples: "\
+                              "https://aka.ms/arg-error-toomanysubs".format(__SUBSCRIPTION_LIMIT)
+            __logger.warning(warning_message)
+
+    response = None
+    try:
+        result_truncated = False
+
+        request_options = QueryRequestOptions(
+            top=first,
+            skip=skip,
+            skip_token=skip_token,
+            result_format=ResultFormat.object_array,
+            allow_partial_scopes=allow_partial_scopes
+        )
+
+        request = QueryRequest(
+            query=graph_query,
+            subscriptions=subs_list,
+            management_groups=mgs_list,
+            options=request_options)
+        response = client.resources(request)  # type: QueryResponse
+        if response.result_truncated == ResultTruncated.true:
+            result_truncated = True
+
+        if result_truncated and first is not None and len(response.data) < first:
+            __logger.warning("Unable to paginate the results of the query. "
```

Comment:
Again, there is no point in logging this to the customer. We need to handle this ourselves